managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -200,7 +210,7 @@ public void addComplete(int rc, final LedgerHandle lh, long entryId, Object ctx)
                     lh == null ? -1 : lh.getId(), entryId, dataLength, rc);
         }
 
-        if (rc != BKException.Code.OK) {
+        if (rc != BKException.Code.OK || timeoutTriggered.isTrue()) {
```

Comment:
It will lead to switch to a new ledger?

Reply:
Yes, once a OpAddEntry called `handleAddFailure(lh)`, it will trigger a ledger switch

Reply:
Maybe some addOps just still in the executor queue, and even not executed, it is reasonable to switch to a new ledger?

Reply:
My fault, only executed addops canbe added to pendingAddOps.
But it looks like it will lead to continuous Switching of Ledgers.
Say, there are 1,2,3 AddOp to ledger 1.
If 1 is timeout, the value of `currentLedgerTimeoutTriggered` will set to be true, and switch to a new ledger.
After 2 is finished, the value of `currentLedgerTimeoutTriggered` is true, it will also switch to a new ledger.
Same as 3.

Reply:
> If 1 is timeout, the value of currentLedgerTimeoutTriggered will set to be true, and switch to a new ledger.
After 2 is finished, the value of currentLedgerTimeoutTriggered is true, it will also switch to a new ledger.
Same as 3.

After a ledger switch, the variable `currentLedgerTimeoutTriggered ` will be recreated [here](https://github.com/apache/pulsar/pull/22221/files#diff-f6a849bd8fdb782ef6c17a2e07a2c54c3dc7d1655c00ec3546d5f3b3fc61e970R1596)